WordPress/Classes/ViewRelated/Tips/AppTips.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+import SwiftUI
+import TipKit
+import Combine
+
+enum AppTips {
+    static func initialize() {
+        guard Feature.enabled(.tipKit), #available(iOS 17, *) else { return }
+        do {
+            try Tips.configure([.displayFrequency(.daily)])
+        } catch {
+            DDLogError("Error initializing tips: \(error)")
+        }
+    }
+
+    @available(iOS 17, *)
+    struct SitePickerTip: Tip {
+        let id = "site_picker_tip"
+
+        var title: Text {
+            Text(NSLocalizedString("tips.sitePickerTip.title", value: "Your Sites", comment: "Tip for site picker"))
+        }
+
+        var message: Text? {
+            Text(NSLocalizedString("tips.sitePickerTip.message", value: "Tap to select a different site or create a new one", comment: "Tip for site picker"))
+        }
+
+        var image: Image? {
+            Image(systemName: "rectangle.stack.badge.plus")
+        }
+
+        @Parameter(.transient)
+        static var blogCount: Int = 0
+
+        var rules: [Rule] {
+            #Rule(Self.$blogCount) { $0 > 1 }
+        }
+
+        var options: [any TipOption] {
+            MaxDisplayCount(1)
+        }
+    }
+}
+
+extension UIViewController {
+    /// Registers a popover to be displayed for the given tip.
+    @available(iOS 17, *)
+    func registerTipPopover(
+        _ tip: some Tip,
+        sourceView: UIView,
+        arrowDirection: UIPopoverArrowDirection? = nil,
+        actionHandler: ((Tips.Action) -> Void)? = nil
+    ) -> TipObserver? {
+        guard Feature.enabled(.tipKit) else {
+            return nil
+        }
+        let task = Task { @MainActor [weak self] in
+            for await shouldDisplay in tip.shouldDisplayUpdates {
```

Comment:
I think you'll need a `if Task.isCancelled { break }` in this for-loop? `task.cancel()` alone probably won't stop the loop.

Reply:
This code is from the Apple [sample code](https://developer.apple.com/documentation/tipkit/tipuipopoverviewcontroller). I'm not sure if it's clearly documented anywhere, but `tip.shouldDisplayUpdates` async sequence (`AsyncMapSequence`) receives cancellation from the task, aka cooperative cancellation.

Reply:
Ah okay. That's quite possible. Ignore my comments then, if the task can be properly cancelled. 🙂